README.md

```diff
@@ -1 +1,961 @@
+[Logs]    [9/17/2019, 11:06:42 AM] [main] Hello world!
+```
+We're back to no debug logs again!
+
+The balenaCLI also allows you to rename environment variables, as well as set
+and remove them for specific devices instead of an entire application fleet.
+For device-based variables, most commands take an extra switch, `--device`
+which allows you to specify the UUID of the device you wish to make the
+variable change for.
+
+# 5. Preloading and Preregistering
+
+Whilst for many cases provisoining a device, moving it to an installation
+location and then connecting it to a stable network with Internet access is
+fairly easy, there are times where a device may have to operate with a limited
+connection (for example a slow GSM connection). In these cases, installing
+a device and then attempting to download an initial application is not only
+slow but sometimes unachievable.
+
+The way round this is to preload an application into an image that is then
+used to provision a device. Preloading injects the images required for
+the services that comprise the application, meaning that on device startup
+the Supervisor can immediately start running those services without having
+to first download the images from the balena registries.
+
+Preloading on it's own is a useful feature for ensuring devices are
+ready to start executing an application as soon as they're powered on, but for
+users who also want to ensure that they know about the devices that they're
+shipping it's only half the story.
+
+Preregistering a device allows the creation of individual registered to an
+application before those devices are ever physically powered on or connected
+to a network. This is extremely useful in situations such as manufacturing
+where a device may require tracking, as a specific device UUID can then be
+associated with a specific customer order, for example. This ensures that
+a customer then receives a device that already has information available
+for it, for support, application updates, etc.
+
+The following exercises will show you how to both preload and preregister
+devices.
+
+## 5.5 Preloading a Device Image
+
+Note that `balena preload` actually uses a Docker container to carry out the
+actual preloading. This is to allow the system to be portable and run under
+Linux, macOS and Windows. As such, you'll need to ensure you've installed
+Docker on your development machine before going any further.
+
+We'll take the current `cliApp` application and preload it the
+`balena-fin-image.img` image we earlier configured and provisioned our device
+with. The `balena preload` command has a large number of switch options for
+catering with different situations, including the ability to use a particular
+release. However, for this exercise, we'll simply use the latest version of
+the code we previously pushed to `cliApp`. In a terminal, execute the following:
+>>>>> NOTE THAT THIS DOESN'T WORK ON MACOS, AS WE SEE AN ERROR WHEN ATTEMPTING
```

Comment:
Is this error because docker for MacOS doesn't support AUFS?

Reply:
I need to look into this in a bit more detail. If we're assuming AUFS, then definitely, but I don't think I've ever managed to get it working under macOS.

Reply:
Yeah, I found out why. When Docker did a major update change, it wiped previously shared directories. I've had to re-add my `/Work` directory. Sigh.